core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java

```diff
@@ -385,11 +385,44 @@ private void initHMC() {
       metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER,
           "io.snappydata.jdbc.EmbeddedDriver");
 
-      try {
-        HiveMetaStoreClient hmc = new HiveMetaStoreClient(metadataConf);
-        SnappyHiveCatalog.this.hmClients.set(hmc);
-      } catch (MetaException me) {
-        throw new IllegalStateException(me);
+      final short numRetries = 3;
+      short count = 0;
+      while (true) {
+        try {
+          HiveMetaStoreClient hmc = new HiveMetaStoreClient(metadataConf);
+          SnappyHiveCatalog.this.hmClients.set(hmc);
+          return;
+        } catch (Exception ex) {
+          Throwable t = ex;
+          boolean noDataStoreFound = false;
+          while (t != null && !noDataStoreFound) {
+            noDataStoreFound = (t instanceof SQLException) &&
+                SQLState.NO_DATASTORE_FOUND.startsWith(
+                    ((SQLException)t).getSQLState());
+            t = t.getCause();
+          }
+          // wait for some time and retry if no data store found error
+          // is thrown due to region not being initialized
+          if (count < numRetries && noDataStoreFound) {
+            try {
+              Misc.getI18NLogWriter().warning(LocalizedStrings.DEBUG,
```

Comment:
slf4j logger should be used as standard for snappydata. This allows us to control logging module-wise easily in log4j.properties (plan is also to add a procedure/SET to change dynamically).